### DISCUSSION TITLE
how to implement personal_sign via Wagmi?

### DISCUSSION THREAD

**Discussion by mathesond2** [wevm/wagmi#692] [General]
Hi, I'm currently tasked with switching over to Rainbowkit from a previous implementation and cannot find out (through the docs, source code or repo) how to pass in the `personal_sign` string to denote the specific type of signature. I was under the impression I could use `useSignMessage` as spoken to in [this Rainbowkit discussion](https://github.com/rainbow-me/rainbowkit/discussions/473) but after some trial/error and going through the wagmi source code, I'm not seeing anything. Any thoughts?

**Comment by beebe4** [comment]
I have a different requirement: 2 of my users have recently reporting issues using Ledgers and on a previous project [this change](https://github.com/purefinance/pure.finance/pull/86) enabled Ledger devices to work successfully.  For the record, I cannot reproduce it because my Ledger is working fine.

**Comment by tmm** [comment]
> Hi, I'm currently tasked with switching over to Rainbowkit from a previous implementation and cannot find out (through the docs, source code or repo) how to pass in the `personal_sign` string to denote the specific type of signature. I was under the impression I could use `useSignMessage` as spoken to in [this Rainbowkit discussion](https://github.com/rainbow-me/rainbowkit/discussions/473) but after some trial/error and going through the wagmi source code, I'm not seeing anything. Any thoughts?

`useSignMessage` is what you want to use.

**Reply by beebe4** [reply]
Is that true for my question as well?

**Reply by tmm** [reply]
Not sure. wagmi does not use `web3.js`.
